internal/syncer/connector/google_cloud_storage.go

```
@@ -0,0 +1,467 @@
+		batchSize:     batchSize,
+		sizeThreshold: sizeThreshold,
+	}, nil
+}
+
+// Validate validates Google Cloud Storage connector settings and credentials.
+func (c *GoogleCloudStorageConnector) Validate(ctx context.Context) error {
+	if c == nil {
+		return fmt.Errorf("Google Cloud Storage connector is nil")
+	}
+	if c.bucketName == "" {
+		return fmt.Errorf("No bucket name was provided in connector settings.")
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)